homeassistant/components/swiss_public_transport/config_flow.py

```diff
@@ -63,25 +94,97 @@ async def async_step_user(
                     user_input[CONF_DESTINATION],
                     session,
                     via=user_input.get(CONF_VIA),
+                    time=user_input.get(CONF_TIME),
                 )
-                try:
-                    await opendata.async_get_data()
-                except OpendataTransportConnectionError:
-                    errors["base"] = "cannot_connect"
-                except OpendataTransportError:
-                    errors["base"] = "bad_config"
-                except Exception:  # pylint: disable=broad-except
-                    _LOGGER.exception("Unknown error")
-                    errors["base"] = "unknown"
+                err = await self.fetch_connections(opendata)
+                if err:
+                    errors["base"] = err
                 else:
-                    return self.async_create_entry(
-                        title=unique_id,
-                        data=user_input,
+                    if user_input[CONF_TIME_MODE] == "now":
+                        unique_id = unique_id_from_config(user_input)
+                        await self.async_set_unique_id(unique_id)
+                        self._abort_if_unique_id_configured()
+                        return self.async_create_entry(
+                            title=unique_id,
+                            data=user_input,
+                        )
+                    self.user_input = user_input
+                    return self.async_show_form(
+                        step_id="advanced",
+                        data_schema=self.build_advanced_schema(user_input),
+                        errors=errors,
+                        description_placeholders=PLACEHOLDERS,
                     )
 
         return self.async_show_form(
             step_id="user",
-            data_schema=DATA_SCHEMA,
+            data_schema=self.add_suggested_values_to_schema(
+                data_schema=USER_DATA_SCHEMA,
+                suggested_values=user_input or {CONF_TIME_MODE: DEFAULT_TIME_MODE},
+            ),
             errors=errors,
             description_placeholders=PLACEHOLDERS,
         )
+
+    async def async_step_advanced(
+        self, advanced_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Async advanced step to set up the connection."""
+        errors: dict[str, str] = {}
+        if advanced_input is not None:
+            unique_id = unique_id_from_config({**self.user_input, **advanced_input})
+            await self.async_set_unique_id(unique_id)
+            self._abort_if_unique_id_configured()
+
+            session = async_get_clientsession(self.hass)
+            time_offset_dict: dict[str, int] | None = advanced_input.get(
+                CONF_TIME_OFFSET
+            )
+            time_offset = (
+                dict_duration_to_str_duration(time_offset_dict)
```

Comment:
Why do we convert this to a string. If I see it right we then convert it to a timedelta? We can just store it as the dict and use that directly to create a timedelta.

Reply:
Yes we can, it was primarily used for unique id, but only needed there.